src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs

```diff
@@ -351,32 +348,47 @@ public Stream InputStream
 
         public Guid RequestTraceIdentifier => Guid.Empty;
 
-        public IAsyncResult BeginGetClientCertificate(AsyncCallback requestCallback, object state)
+        private IAsyncResult BeginGetClientCertificateCore(AsyncCallback requestCallback, object state)
         {
-            Task<X509Certificate2> getClientCertificate = new Task<X509Certificate2>(() => GetClientCertificate());
-            return TaskToApm.Begin(getClientCertificate, requestCallback, state);
+            var asyncResult = new GetClientCertificateAsyncResult(this, state, requestCallback);
+
+            // The certificate is already retrieved by the time this method is called. GetClientCertificate() evaluates to
+            // a simple member access, so this will always complete immediately.
```

Comment:
This comment isn't entirely true.  GetClientCertificateCore() evaluates to simple member accesses... but GetClientCertificate does more than that.  Which is supposed to be called here?